compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1128,18 +1160,34 @@ setRootConfig(func::FuncOp entryPointFn,
     maxTileSizes[0] = 1;
   }
 
+  // Compute cache-level tile sizes. Cache a dimension only if there are
+  // enough iterations.
+  SmallVector<int64_t> cacheTileSizes;
+  cacheTileSizes = getDefaultMatmulCacheSizes(linalgOp, isQuantized);
+  cacheTileSizes = getMatmulCacheTileSizesForShape(
+      cacheTileSizes, linalgOp.getStaticLoopRanges());
+
+  // Choose the next non-zero tile size immediately after the distribution
+  // level to help compute the distribution tile sizes.
+  SmallVector<int64_t> minTileSizes;
+  for (auto [cacheTileSize, vecTileSize] :
+       llvm::zip_equal(cacheTileSizes, vecTileSizes)) {
+    int64_t minTileSize = cacheTileSize != 0 ? cacheTileSize : vecTileSize;
```

Comment:
This can be simplified to a common form: `A ? A : B;`

```suggestion
    int64_t minTileSize = cacheTileSize ? cacheTileSize : vecTileSize;
```

Reply:
Via an GCC extension you can do `A ?: B`  https://gcc.gnu.org/onlinedocs/gcc/Conditionals.html :P

Though personally I think the explicit `!= 0` check is a little clearer.

Reply:
I also like to make the int -> bool conversion explicit as I think it's easier to read. Thanks for the suggestion, though!